options/options_test.go

```diff
@@ -295,6 +295,42 @@ func TestParseAndSetOptions(t *testing.T) {
 				},
 				ShouldError: false,
 			},
+			{
+				Name: "aws auth fields set",
+				CS: connstring.ConnString{
+					AuthMechanism: "MONGODB-AWS",
+					AuthSource:    "",
+					AuthSourceSet: true,
+					PasswordSet:   true,
+				},
+				OptsIn: &ToolOptions{
+					General:        &General{},
+					Verbosity:      &Verbosity{},
+					Connection:     &Connection{},
+					URI:            &URI{},
+					SSL:            &SSL{},
+					Auth:           &Auth{},
+					Namespace:      &Namespace{},
+					Kerberos:       &Kerberos{},
+					enabledOptions: EnabledOptions{Auth: true, URI: true},
+				},
+				OptsExpected: &ToolOptions{
+					General:    &General{},
+					Verbosity:  &Verbosity{},
+					Connection: &Connection{},
+					URI:        &URI{},
+					SSL:        &SSL{},
+					Auth: &Auth{
+						Source:    "",
+						Mechanism: "MONGODB-AWS",
+						AWSSessionToken: "token",
```

Comment:
I don't understand how this works. Shouldn't you have to set `CS.AuthMechanismProperties["AWS_SESSION_TOKEN"] = "token"` for this to work?

Reply:
You're right -- I didn't notice that I had to add a specific assert for this AWS auth test (at the very end). The test failed when I added in the correct assert, and passed once I added ` CS.AuthMechanismProperties["AWS_SESSION_TOKEN"] = "token"`.